examples/flutter_gallery/lib/demo/rally/charts/pie_chart.dart

```diff
@@ -0,0 +1,249 @@
+/// have empty space.
+class RallyPieChart extends StatefulWidget {
+  const RallyPieChart(
+      {this.heroLabel, this.heroAmount, this.wholeAmount, this.segments});
+
+  final String heroLabel;
+  final double heroAmount;
+  final double wholeAmount;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  _RallyPieChartState createState() => _RallyPieChartState();
+}
+
+class _RallyPieChartState extends State<RallyPieChart>
+    with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+        duration: const Duration(milliseconds: 600), vsync: this);
+    animation = CurvedAnimation(
+        parent: TweenSequence<double>(<TweenSequenceItem<double>>[
+          TweenSequenceItem<double>(
+            tween: Tween<double>(begin: 0.0, end: 0.0),
+            weight: 1.0,
+          ),
+          TweenSequenceItem<double>(
+            tween: Tween<double>(begin: 0.0, end: 1.0),
+            weight: 1.5,
+          ),
+        ]).animate(controller),
+        curve: Curves.decelerate);
+    controller.forward();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _AnimatedRallyPieChart(
+      animation: animation,
+      centerLabel: widget.heroLabel,
+      centerAmount: widget.heroAmount,
+      total: widget.wholeAmount,
+      segments: widget.segments,
+    );
+  }
+}
+
+class _AnimatedRallyPieChart extends AnimatedWidget {
+  const _AnimatedRallyPieChart({
+    Key key,
+    this.animation,
+    this.centerLabel,
+    this.centerAmount,
+    this.total,
+    this.segments,
+  }) : super(key: key, listenable: animation);
+
+  final Animation<double> animation;
+  final String centerLabel;
+  final double centerAmount;
+  final double total;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle labelTextStyle = Theme.of(context)
+        .textTheme
+        .body1
+        .copyWith(fontSize: 14.0, letterSpacing: 0.5);
+
+    return DecoratedBox(
+      decoration: _RallyPieChartOutlineDecoration(
+          maxFraction: animation.value, total: total, segments: segments),
+      child: SizedBox(
+        height: 300.0,
+        child: Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              Text(
+                centerLabel,
+                style: labelTextStyle,
+              ),
+              Text(
+                usdWithSignFormat.format(centerAmount),
+                style: Theme.of(context).textTheme.headline,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _RallyPieChartOutlineDecoration extends Decoration {
+  const _RallyPieChartOutlineDecoration(
```

Comment:
const _RallyPieChartOutlineDecoration({ this.maxFraction, this.total, this.segments });